gluten-core/src/main/scala/org/apache/gluten/execution/WindowExecTransformer.scala

```diff
@@ -67,14 +67,10 @@ case class WindowExecTransformer(
   }
 
   override def requiredChildOrdering: Seq[Seq[SortOrder]] = {
-    if (
-      BackendsApiManager.getSettings.requiredChildOrderingForWindow()
-      && GlutenConfig.getConf.veloxColumnarWindowType.equals("streaming")
-    ) {
-      // Velox StreamingWindow need to require child order.
-      Seq(partitionSpec.map(SortOrder(_, Ascending)) ++ orderSpec)
+    if (GlutenConfig.getConf.veloxColumnarWindowType.equals("sort")) {
+      super.requiredChildOrdering
     } else {
-      Seq(Nil)
+      Seq(partitionSpec.map(SortOrder(_, Ascending)) ++ orderSpec)
```

Comment:
There seems have two problems, first, super.requiredChildOrdering = Seq(partitionSpec.map(SortOrder(_, Ascending)) ++ orderSpec), in addition, according to the original logic, ck backend should not need child order.

Reply:
for 1, it's a good point, it seems the `WindowExecBase` has implemented these methods since 3.3
for 2, the ck backend requires child ordering and before we did not remove local sort before window

Reply:
I think we should retain requiredChildOrderingForWindow instead of using velox configuration to control ck backend  
although it has no effect by default value.